aeron-system-tests/src/test/java/io/aeron/TaggedFlowControlSystemTest.java

```diff
@@ -344,23 +344,13 @@ void shouldPreventConnectionUntilRequiredGroupSizeMatchTagIsMet()
 
                 publication = clientA.addPublication(uriWithTaggedFlowControl, STREAM_ID);
 
-                Thread.sleep(100);
-                assertFalse(publication.isConnected());
-
                 subscription0 = clientA.addSubscription(uriPlain, STREAM_ID);
-
-                Thread.sleep(100);
-                assertFalse(publication.isConnected());
-
                 subscription1 = clientA.addSubscription(uriPlain, STREAM_ID);
                 subscription2 = clientA.addSubscription(uriPlain, STREAM_ID);
-
-                Thread.sleep(100);
-                assertFalse(publication.isConnected());
-
                 subscription3 = clientB.addSubscription(uriWithReceiverTag, STREAM_ID);
 
-                Thread.sleep(100);
+                // Sleep for a bit to ensure that we haven't become connected.
+                Thread.sleep(500);
```

Comment:
Need to find a better way than this.

Reply:
It's tricky testing to see if something hasn't happened.  Perhaps I can look at some of the counters to infer that the NetworkPublication has see status messages from 2 endpoints.

Reply:
I've got rid of all of the long timeouts in the flow control tests.  Mostly I'm waiting for the subscription to connect and seeing the status messages increment directly after that.